polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -798,8 +840,17 @@ void LoopInternalization::transform(T loop,
   std::variant<Value, unsigned> offset =
       ValueOrUnsigned::get(0, builder, workGroupSize.hasElemTy<Value>());
   for (Operation *memref : memrefs) {
-    sycl::AccessorType accTy =
-        getAccessorType(cast<sycl::SYCLAccessorSubscriptOp>(memref));
+    auto accSub = cast<sycl::SYCLAccessorSubscriptOp>(memref);
+    sycl::AccessorType accTy = getAccessorType(accSub);
+    const auto idTy = cast<sycl::IDType>(
+        cast<sycl::AccessorImplDeviceType>(accTy.getBody()[0]).getBody()[0]);
+    // TODO: The current indexes used are incorrect.
+    SmallVector<Value> indexes(localIDs.begin(), localIDs.end());
```

Comment:
Actually, I would pass localIDs to this function rather than adding the SmallVector to the class. That way there would be no need to clear it for every kernel.

Reply:
done.